plugins/SitesManager/API.php

```diff
@@ -394,6 +394,24 @@ public function getSitesWithAdminAccess($fetchAliasUrls = false, $pattern = fals
         return $sites;
     }
 
+    /**
+     * Returns the messages to warn users on site deletion.
+     *
+     * @param int $idSite
+     * @return array messages to warn users
+     * @throws Exception if the website ID doesn't exist or the user doesn't have super user access to it
+     * @internal
+     * @unsanitized
+     */
+    public function getMessagesToWarnOnSiteRemoval(int $idSite): array
+    {
+        $messages = [];
+        Piwik::checkUserHasSuperUserAccess($idSite);
+        Piwik::postEvent('SitesManager.getMessagesToWarnOnSiteRemoval', [$idSite, &$messages]);
```

Comment:
We should add a comment above the event to document what it is good for. Like it'S e.g. the case here:
https://github.com/matomo-org/matomo/blob/9a3ef94df61682e7d543f3f8b438c368f6eb71a6/core/Scheduler/Scheduler.php#L157-L166

And when looking at other events it seems we use to pass the referenced parameter first, and then the other. So you may switch `idSite` and `messages`

Reply:
@sgiehl Added as suggested :+1: